src/avram/needy_initializer_and_delete_methods.cr

```diff
@@ -124,11 +124,8 @@ module Avram::NeedyInitializerAndDeleteMethods
       {% if with_bang %}
         operation.delete!
       {% else %}
-        if operation.delete
-          yield operation, operation.record.not_nil!
-        else
-          yield operation, nil
-        end
+        operation.delete
+        yield operation, operation.record.not_nil!
```

Comment:
```suggestion
        yield operation, operation.record
```

I don't believe you even need `not_nil!`. It doesn't look like it's defined optionally in `DeleteOperation`. Could be wrong, though.